pinn/networks/pinet2_splitting.py

```diff
@@ -0,0 +1,356 @@
+            out_pool (str): pool atomic outputs, see ANNOutput
+            depth (int): number of interaction blocks
+            rc (float): cutoff radius
+            basis_type (string): basis function, can be "polynomial" or "gaussian"
+            n_basis (int): number of basis functions to use
+            gamma (float or array): width of gaussian function for gaussian basis
+            center (float or array): center of gaussian function for gaussian basis
+            cutoff_type (string): cutoff function to use with the basis.
+            act (string): activation function to use
+            weighted (bool): whether to use weighted style
+        """
+        super(PiNet2Splitting, self).__init__()
+
+        self.depth = depth
+        self.preprocess = PreprocessLayer(atom_types, rc)
+        self.cutoff = CutoffFunc(rc, cutoff_type)
+
+        if basis_type == "polynomial":
+            self.basis_fn = PolynomialBasis(n_basis)
+        elif basis_type == "gaussian":
+            self.basis_fn = GaussianBasis(center, gamma, rc, n_basis)
+
+        self.res_update1 = [ResUpdate() for i in range(depth)]
+        self.res_update3 = [ResUpdate() for i in range(depth)]
+        self.gc_blocks = [
+            GCBlock(weighted, pp_nodes, pi_nodes, ii_nodes, activation=act)
+            for i in range(depth)
+        ]
+        self.out_layers = [OutLayer(out_nodes, out_units) for i in range(depth)]
+        self.ann_output = ANNOutput(out_pool)
+
+    def call(self, tensors):
+        """PiNet takes batches atomic data as input, the following keys are
+        required in the input dictionary of tensors:
+
+        - `ind_1`: [sparse indices](layers.md#sparse-indices) for the batched data, with shape `(n_atoms, 1)`;
+        - `elems`: element (atomic numbers) for each atom, with shape `(n_atoms)`;
+        - `coord`: coordintaes for each atom, with shape `(n_atoms, 3)`.
+
+        Optionally, the input dataset can be processed with
+        `PiNet.preprocess(tensors)`, which adds the following tensors to the
+        dictionary:
+
+        - `ind_2`: [sparse indices](layers.md#sparse-indices) for neighbour list, with shape `(n_pairs, 2)`;
+        - `dist`: distances from the neighbour list, with shape `(n_pairs)`;
+        - `diff`: distance vectors from the neighbour list, with shape `(n_pairs, 3)`;
+        - `prop`: initial properties `(n_pairs, n_elems)`;
+
+        Args:
+            tensors (dict of tensors): input tensors
+
+        Returns:
+            output (tensor): output tensor with shape `[n_atoms, out_nodes]`
+        """
+        tensors = self.preprocess(tensors)
+        tensors["p3"] = tf.zeros([tf.shape(tensors["ind_1"])[0], 3, 1])
+        fc = self.cutoff(tensors["dist"])
+        basis = self.basis_fn(tensors["dist"], fc=fc)
+        output = 0.0
+        for i in range(self.depth):
+            p1, p3 = self.gc_blocks[i](
+                [tensors["ind_2"], tensors["p1"], tensors["p3"], tensors["diff"], basis]
```

Comment:
Here we should replace the tensors["diff"] to tensors["norm_diff"]

Reply:
those architecture will be remove except pinet2 & pinet2_modularized